hnix-store-core/src/System/Nix/ReadonlyStore.hs

```diff
@@ -69,3 +69,22 @@ makeFixedOutputPath fp recursive h =
 computeStorePathForText
   :: FilePath -> StorePathName -> ByteString -> (StorePathSet -> StorePath)
 computeStorePathForText fp nm = makeTextPath fp nm . hash
+
+computeStorePathForPath
+  :: StorePathName        -- ^ Name part of the newly created `StorePath`
+  -> FilePath             -- ^ Local `FilePath` to add
+  -> Bool                 -- ^ Add target directory recursively
+  -> (FilePath -> Bool)   -- ^ Path filter function
+  -> Bool                 -- ^ Only used by local store backend
```

Comment:
I feel like we should do something about these two arguments but I'm not sure what.